app/models/queue_item.rb

```diff
@@ -1,10 +1,38 @@
 class QueueItem < ActiveRecord::Base
+  validate :position_cannot_be_less_than_one
+  validate :position_cannot_be_greater_than_queue_items_count
+
   belongs_to :user
   belongs_to :video
 
   delegate :category, to: :video
   delegate :title, to: :video, prefix: :video
 
+  def self.save_and_update_positions(queue_ids_with_positions)
+    begin
+      ActiveRecord::Base.transaction do
+        queue_ids_with_positions.each do |id, position|
+            queue_item = QueueItem.find(id.to_i)
+            queue_item.update!(position: position)
+        end
+      end
+    rescue ActiveRecord::RecordInvalid => e
+      return e.message
+    end
+  end
+
+  def position_cannot_be_less_than_one
+    if !position.nil? && position < 1
+      errors.add(:position, "can't be less than 1")
+    end
+  end
+
+  def position_cannot_be_greater_than_queue_items_count
```

Comment:
I am not sure if this is the best way to handle this case. I would prefer to set position to 1000000 if i would have lots of items and would like to push one to the bottom.